js/src/edit-block/hooks/useField.js

```diff
@@ -276,12 +277,20 @@ const useField = () => {
 
 		if ( newSettings.hasOwnProperty( 'name' ) ) {
 			if ( hasParent ) {
+				// If the new name already exists like 'foo', append a space like 'foo '.
+				// The block.fields object is keyed by field.name, so it can't have duplicate names.
+				newSettings.name = newBlock.fields[ fieldToChange.parent ].sub_fields[ newSettings.name ]
+					? `${ newSettings.name } `
+					: newSettings.name;
```

Comment:
This is a little hacky. But it's much simpler than [previous](https://github.com/studiopress/genesis-custom-blocks/pull/56/commits/c6e7cc62a2205f65fe884d99fb0c19ba82ecda9c) [attempts](https://github.com/studiopress/genesis-custom-blocks/pull/56/commits/3cab15fb343b681a388f578f077218da820f6d07).

The problem is that no 2 fields can have the same name.

This is because fields are keyed by name in the `block.fields`:

```js
{
        app: {location: "editor", width: "100", help: "", default: "", placeholder: "", …},
       "app name": {location: "editor", width: "100", help: "", default: "", placeholder: "", …},
}
```

But when entering field names, users might temporarily have 2 fields with the same name. 

Like:

![field-will-be-app](https://user-images.githubusercontent.com/4063887/105552434-85ecc280-5cc9-11eb-830d-c896d88c3fcf.gif)

So when 2 fields would have the same name, this appends a space to the duplicate, like `"app "`:

```js
{
        app: {location: "editor", width: "100", help: "", default: "", placeholder: "", …},
       "app ": {location: "editor", width: "100", help: "", default: "", placeholder: "", …},
}
```

Overall, it's a good thing that this object doesn't allow 2 of the same name.

`block_field( 'app' )` should only have to look at 1 field, not 2.